docs/source/newsfragments/4396.change.rst

```diff
@@ -0,0 +1 @@
+The attribute ``frequency`` on :class:`~cocotb.clock.Clock` was removed, and the attribute :attr:`~cocotb.clock.Clock.preiod` was change to be in terms of the passed in units rather than simulation steps.
```

Comment:
```suggestion
The attribute ``frequency`` on :class:`~cocotb.clock.Clock` was removed, and the attribute :attr:`~cocotb.clock.Clock.period` was changed to be in terms of the passed-in units rather than simulation steps.
```